exchanges/request/limit.go

```diff
@@ -42,7 +42,7 @@ type EndpointLimit int
 // wrapper for extended rate limiting configuration i.e. Shells of rate
 // limits with a global rate for sub rates.
 type Limiter interface {
-	Limit(context.Context, EndpointLimit) error
+	Limit(context.Context, EndpointLimit) (*rate.Limiter, int, error)
```

Comment:
Signature could potentially even be a uint16, stronger typed and we can never have negative numbers assigned